libbeat/processors/communityid/communityid.go

```diff
@@ -0,0 +1,265 @@
+	return event, err
+}
+
+func (p *processor) buildFlow(event *beat.Event) *flowhash.Flow {
+	var flow flowhash.Flow
+
+	// source ip
+	v, err := event.GetValue(p.Fields.SourceIP)
+	if err != nil {
+		return nil
+	}
+	var ok bool
+	flow.SourceIP, ok = tryToIP(v)
+	if !ok {
+		return nil
+	}
+
+	// destination ip
+	v, err = event.GetValue(p.Fields.DestinationIP)
+	if err != nil {
+		return nil
+	}
+	flow.DestinationIP, ok = tryToIP(v)
+	if !ok {
+		return nil
+	}
+
+	// protocol
+	v, err = event.GetValue(p.Fields.TransportProtocol)
+	if err != nil {
+		return nil
+	}
+	flow.Protocol, ok = tryToIANATransportProtocol(v)
+	if !ok {
+		return nil
+	}
+
+	switch flow.Protocol {
+	case tcpProtocol, udpProtocol, sctpProtocol:
+		// source port
+		v, err = event.GetValue(p.Fields.SourcePort)
+		if err != nil {
+			return nil
+		}
+		flow.SourcePort, ok = tryToUint16(v)
+		if !ok || flow.SourcePort == 0 {
+			return nil
+		}
+
+		// destination port
+		v, err = event.GetValue(p.Fields.DestinationPort)
+		if err != nil {
+			return nil
+		}
+		flow.DestinationPort, ok = tryToUint16(v)
+		if !ok || flow.DestinationPort == 0 {
+			return nil
+		}
+	case icmpProtocol, icmpIPv6Protocol:
+		v, err = event.GetValue(p.Fields.ICMPType)
+		if err != nil {
+			return nil
+		}
+		flow.ICMP.Type, ok = tryToUint8(v)
+		if !ok {
+			return nil
+		}
+
+		v, err = event.GetValue(p.Fields.ICMPCode)
+		if err != nil {
+			return nil
+		}
+		flow.ICMP.Code, ok = tryToUint8(v)
+		if !ok {
+			return nil
+		}
+	}
+
+	return &flow
+}
+
+func tryToIP(from interface{}) (net.IP, bool) {
+	switch v := from.(type) {
+	case net.IP:
+		return v, true
+	case string:
+		ip := net.ParseIP(v)
+		return ip, ip != nil
+	default:
+		return nil, false
+	}
+}
+
+// tryToUint16 tries to coerce the given interface to an uint16. On success it
+// returns the int value and true.
+func tryToUint16(from interface{}) (uint16, bool) {
+	switch v := from.(type) {
+	case int:
+		return uint16(v), true
+	case int8:
+		return uint16(v), true
+	case int16:
+		return uint16(v), true
+	case int32:
+		return uint16(v), true
+	case int64:
+		return uint16(v), true
+	case uint:
+		return uint16(v), true
+	case uint8:
+		return uint16(v), true
+	case uint16:
+		return v, true
+	case uint32:
+		return uint16(v), true
+	case uint64:
+		return uint16(v), true
+	case string:
+		num, err := strconv.ParseUint(v, 0, 16)
+		if err != nil {
+			return 0, false
+		}
+		return uint16(num), true
+	default:
+		return 0, false
+	}
+}
+
+func tryToUint8(from interface{}) (uint8, bool) {
+	to, ok := tryToUint16(from)
+	return uint8(to), ok
+}
+
+const (
+	icmpProtocol     uint8 = 1
+	igmpProtocol     uint8 = 2
+	tcpProtocol      uint8 = 6
+	udpProtocol      uint8 = 17
+	icmpIPv6Protocol uint8 = 58
+	sctpProtocol     uint8 = 132
+)
+
+var transports = map[string]uint8{
+	"icmp":      icmpProtocol,
+	"igmp":      igmpProtocol,
+	"tcp":       tcpProtocol,
+	"udp":       udpProtocol,
+	"icmp-ipv6": icmpIPv6Protocol,
```

Comment:
add `ipv6-icmp` (IANA, ECS docs) and `icmpv6` (IPv6 rfc)

Reply:
I meant to use the IANA value. So I fixed that mistake and added `icmpv6`.